pkg/errors/error.go

```diff
@@ -50,22 +50,35 @@ const (
 	errDatabase     = errClass("database")
 )
 
+func isCodeCancellation(code gcodes.Code) bool {
+	return code == gcodes.Canceled || code == gcodes.DeadlineExceeded
```

Comment:
In other places we use a map, but since it's only 2 values this feels simpler to me.
Happy to change it if needed